suite-native/module-add-coin/src/hooks/useAddCoinAccount.ts

```diff
@@ -0,0 +1,250 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { useNavigation } from '@react-navigation/native';
+
+import { TrezorDevice } from '@suite-common/suite-types';
+import { AccountType, NetworkSymbol, Network } from '@suite-common/wallet-config';
+import {
+    AccountsRootState,
+    DeviceRootState,
+    selectDevice,
+    selectDeviceAccounts,
+} from '@suite-common/wallet-core';
+import { Account } from '@suite-common/wallet-types';
+import { useAlert } from '@suite-native/alerts';
+import {
+    addNetworkAccountThunk,
+    selectAreTestnetsEnabled,
+    selectDiscoverySupportedNetworks,
+    getCardanoSupportedAccountTypesThunk,
+} from '@suite-native/discovery';
+import { useTranslate } from '@suite-native/intl';
+import { useOpenLink } from '@suite-native/link';
+import {
+    RootStackParamList,
+    RootStackRoutes,
+    StackNavigationProps,
+} from '@suite-native/navigation';
+
+const availableNetworkAccountTypes: Map<NetworkSymbol | undefined, AccountType[]> = new Map([
+    ['btc', ['normal', 'taproot', 'segwit', 'legacy']],
+    ['ltc', ['normal', 'segwit', 'legacy']],
+    ['btg', ['normal', 'segwit', 'legacy']],
+    ['dgb', ['normal', 'segwit', 'legacy']],
+    ['vtc', ['normal', 'segwit', 'legacy']],
+]);
```

Comment:
IMO would be better to define this as an object with type `as const satisfies Record<NetworkSymbol, AccountType[]>`. Like that the TypeScript will warn you if there will be a new coin added in the future but it would not be handled here. Potentionally migh save us from some critical bugs in the future.